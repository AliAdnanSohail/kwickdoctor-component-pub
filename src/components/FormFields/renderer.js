import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
