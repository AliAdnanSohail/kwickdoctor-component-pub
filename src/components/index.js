export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
