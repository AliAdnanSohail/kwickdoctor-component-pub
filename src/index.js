export * from './components';

export { default as Validate } from './utils/validations';
