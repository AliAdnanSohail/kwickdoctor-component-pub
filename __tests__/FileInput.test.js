import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { FileInput } from '../src';

it('renders correctly', () => {
  const element = renderer.create(<FileInput id="test">test</FileInput>).toJSON();

  expect(element).toMatchSnapshot();
});

it('Fileinput change correct', () => {
  const wrapper = shallow(<FileInput id="test" />);
  console.log(wrapper.debug());
  wrapper.find('input').simulate('change', {
    target: {
      files: ['someimage.png'],
    },
  });
});
