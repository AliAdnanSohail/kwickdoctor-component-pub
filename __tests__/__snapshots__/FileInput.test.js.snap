// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <input
    className="jsx-1181367161 inputFile"
    id="test"
    name=""
    onChange={[Function]}
    type="file"
  />,
  <label
    className="jsx-1181367161 upload-license-copy"
    htmlFor="test"
  >
    <svg
      className="jsx-1181367161 icon-upload"
      fill="#4695ef"
      height="17"
      viewBox="0 0 20 17"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="jsx-1181367161"
        d="M10 0l-5.2 4.9h3.3v5.1h3.8v-5.1h3.3l-5.2-4.9zm9.3 11.5l-3.2-2.1h-2l3.4 2.6h-3.5c-.1 0-.2.1-.2.1l-.8 2.3h-6l-.8-2.2c-.1-.1-.1-.2-.2-.2h-3.6l3.4-2.6h-2l-3.2 2.1c-.4.3-.7 1-.6 1.5l.6 3.1c.1.5.7.9 1.2.9h16.3c.6 0 1.1-.4 1.3-.9l.6-3.1c.1-.5-.2-1.2-.7-1.5z"
      />
    </svg>
    <span
      className="jsx-1181367161"
    >
      Upload file
    </span>
  </label>,
]
`;
